django_webapps/scripts/db_population/parsers/teacher_parsers.py

```diff
@@ -0,0 +1,175 @@
+import re
+from enum import Enum
+
+from email_validator import EmailNotValidError, validate_email
+
+from api.models.auxil.constants import TIME_SLOTS, LanguageLevelId
+from api.models.choices.status.project import TeacherProjectStatus
+from api.models.choices.status.situational import TeacherSituationalStatus
+
+
+class ProjectStatusAction(str, Enum):
+    SPEAKING_CLUB = "speking_club"
+    CV_AND_INTERVIEW = "cv_and_interview"
+    NONE = "none"
+
+
+def parse_project_status(
+    status_str: str,
+) -> tuple[TeacherProjectStatus, ProjectStatusAction] | None:
+    status_str = status_str.lower()
+    result = None
+    if status_str in ["идут занятия"]:
+        result = (TeacherProjectStatus.WORKING, ProjectStatusAction.NONE)
+    elif status_str in ["перерыв в преподавании"]:
+        result = (TeacherProjectStatus.ON_LEAVE, ProjectStatusAction.NONE)
+    elif status_str in ["закончил участие в проекте"]:
+        result = (TeacherProjectStatus.FINISHED_LEFT, ProjectStatusAction.NONE)
+    elif status_str in ["готов начать позже (см.коммент)"]:  # noqa: SIM114
+        pass
+    elif status_str in ["Substitute Teacher"]:
+        pass
+    elif status_str in ["Speaking Clubs"]:
+        result = (TeacherProjectStatus.WORKING, ProjectStatusAction.SPEAKING_CLUB)
+    elif status_str in ["жду ответа", "ищу учеников"]:
+        result = (TeacherProjectStatus.NO_GROUP_YET, ProjectStatusAction.NONE)
+    elif status_str in ["не сможет участвовать в проекте"]:
+        pass
+    elif status_str in ["CV and Interview"]:
+        result = (TeacherProjectStatus.NO_GROUP_YET, ProjectStatusAction.CV_AND_INTERVIEW)
+    elif status_str in ["дубль"]:  # duplicate?
+        pass
+    return result
+
+
+def parse_situational_status(status_str: str) -> TeacherSituationalStatus | None:
+    status_str = status_str.lower()
+    if status_str in ["не вышел на связь"]:
+        return TeacherSituationalStatus.NO_RESPONSE
+    if status_str in ["группы набраны"]:
+        return TeacherSituationalStatus.AWAITING_START
+    return None
+
+
+def parse_name(name_str: str) -> str:
+    return name_str
+
+
+def parse_email(email_str: str) -> str | None:
+    try:
+        validate_email(email_str)
+    except EmailNotValidError as _:
+        return None
+    return email_str
+
+
+def parse_has_experience(exp_str: str) -> bool:
+    if exp_str.lower().strip() == "да":
+        return True
+    return False
+
+
+def parse_telegram_name(tg_str: str) -> str | None:
+    tg_regex = re.compile(r"@(?=\w{5,32}\b)[a-zA-Z0-9]+(?:_[a-zA-Z0-9]+)*.*")
+    match_result = re.search(tg_regex, tg_str)
+    if match_result is not None:
```

Comment:
`if match_result := re.search(tg_regex, tg_str) is not None:`

then string above could be removed